modern/src/settings/MaintenancePage.js

```diff
@@ -0,0 +1,161 @@
+import React, { useState } from 'react';
+import t from '../common/localization';
+import { prefixString } from '../common/stringUtils';
+import EditItemView from '../EditItemView';
+import { Accordion, AccordionSummary, AccordionDetails, makeStyles, Typography, TextField, FormControl, InputLabel, MenuItem, Select, } from '@material-ui/core';
+import InputAdornment from '@material-ui/core/InputAdornment';
+import ExpandMoreIcon from '@material-ui/icons/ExpandMore';
+import EditAttributesView from '../attributes/EditAttributesView';
+import positionAttributes from '../attributes/positionAttributes';
+import { useAttributePreference } from '../common/preferences';
+import { speedFromKnots, speedToKnots, distanceFromMeters, distanceToMeters } from '../common/converter';
+
+const useStyles = makeStyles(() => ({
+  details: {
+    flexDirection: 'column',
+  },
+}));
+
+const MaintenancePage = () => {
+  
+  const classes = useStyles();
+  const [item, setItem] = useState();
+  const [labels, setLabels] = useState({start: '', period: ''});
+
+  const speedUnit = useAttributePreference('speedUnit');
+  const distanceUnit = useAttributePreference('distanceUnit');
+
+  const convertToList = (attributes) => {
+    let otherList = [];
+    for (const key in attributes) {
+      const value = attributes[key];
+      if (value.type === 'number') {
+        otherList.push({key, name: value.name, type: value.type});
+      }
+    }
+    return otherList;
+  }
+
+  const onMaintenanceTypeChange = event => {
+    const newValue = event.target.value;
+    setItem({...item, type: newValue, start: 0, period: 0});
+
+    const attribute = positionAttributes[newValue];
+    if (attribute && attribute.dataType) {
+      switch (attribute.dataType) {
+        case 'distance':
+          setLabels({ ...labels, start: t(prefixString('shared', distanceUnit)), period: t(prefixString('shared', distanceUnit))});
+          break;
+        case 'speed':
+          setLabels({ ...labels, start: t(prefixString('shared', speedUnit)), period: t(prefixString('shared', speedUnit))});
+          break;
+        default:
+          break;
+      }
+    }
+  }
+
+  const rawToValue = (value) => {
+
+    const attribute = positionAttributes[item.type];
+    if (attribute && attribute.dataType) {
+      switch (attribute.dataType) {
+        case 'speed':
+          return speedFromKnots(value, speedUnit);
+        case 'distance':
+          return distanceFromMeters(value, distanceUnit);
+        default:
```

Comment:
Do we need default if we already have "global" return?

Reply:
Removed.